src/test/java/com/hankcs/hanlp/algorithm/EditDistanceTest.java

```diff
@@ -0,0 +1,57 @@
+package com.hankcs.hanlp.algorithm;
+
+import com.hankcs.hanlp.corpus.synonym.Synonym;
+import com.hankcs.hanlp.dictionary.common.CommonSynonymDictionary;
+import com.hankcs.hanlp.dictionary.common.CommonSynonymDictionary.SynonymItem;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+public class EditDistanceTest {
+
+    @Test
+    public void testComputeCharArray() {
+        Assert.assertEquals(2,
+            EditDistance.compute("foo".toCharArray(), "oof".toCharArray()));
+    }
+
+    @Test
+    public void testComputeString() {
+        Assert.assertEquals(2,
+            EditDistance.compute("foo", "oof"));
+    }
+
+    @Test
+    public void testComputeList() {
+        Assert.assertEquals(32L,
+            EditDistance.compute(
+                new ArrayList<SynonymItem>(Arrays.asList(
+                    new CommonSynonymDictionary.SynonymItem(
+                        new Synonym("", 64L), null, '='))),
+                new ArrayList<SynonymItem>(Arrays.asList(
+                    new CommonSynonymDictionary.SynonymItem(
+                        new Synonym("", 32L), null, '=')))));
```

Comment:
I would do this instead, as it is easier to read and you can do it without `Arrays.asList()`

        ArrayList<SynonymItem> synonymItems1 = new ArrayList<SynonymItem>();
        synonymItems1.add(new SynonymItem(new Synonym("", 32L), null, '='));

        ArrayList<SynonymItem> synonymItems2 = new ArrayList<SynonymItem>();
        synonymItems2.add(new SynonymItem(new Synonym("", 64L), null, '='));
        Assert.assertEquals(32L,
            EditDistance.compute(synonymItems2, synonymItems1));

Reply:
I agree, inlining this is a bit too difficult to read.